File: Level2/짝지어 제거하기.py
s='baabaa'
def solution(s):
    answer=[]
    for i in range(len(s)):
        if not answer:
            answer.append(s[i])
        else:
            if s[i]==answer[-1]:
                answer.pop()
            else:
                answer.append(s[i])
    
    return 1 if len(answer)==0 else 0
print(solution(s))

File: Level2/짝지어 제거하기.cpp
#include <iostream>
#include <vector>
#include<string>
using namespace std;

int solution(string s) {
    int answer = -1;
    vector<char> v;
    for (int i = 0; i < s.length(); i++) {
        if (v.empty())
            v.push_back(s[i]);
        else {
            if (v[v.size() - 1] == s[i])
                v.pop_back();
            else
                v.push_back(s[i]);
        }
    }
    if (v.size() == 0) 
        answer = 1;
    else 
        answer = 0;
    return answer;
}